src/datamanager/core.py

```
@@ -58,35 +58,41 @@ def hash_file(file_path: Path) -> str:
     return h.hexdigest()
 
 
-def upload_to_r2(client: S3Client, file_path: Path, object_key: str) -> None:
+def upload_to_r2(
+    client: S3Client, file_path: Path, object_key: str, bucket: str = None
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)